warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -5684,12 +5684,39 @@ public static List<GeoTimeSerie> map(GeoTimeSerie gts, Object mapper, long prewi
     // Retrieve ticks if GTS is not bucketized.
     long[] ticks = isBucketized(gts) ? null : Arrays.copyOf(mapped.ticks, gts.values);
 
-    // Sort outputTicks
+    // Check if outputTicks is correctly sorted
+    // In case of a concurrent execution, sorting outputTicks here will lead to a ConcurrentModificationException
     if (null != outputTicks) {
-      if (reversed) {
-        Collections.sort(outputTicks, Collections.<Long>reverseOrder());
-      } else {
-        Collections.sort(outputTicks);
+      if (outputTicks.size() > 1) {
+        if (reversed) {
+          boolean descending = true;
+          int i = 1;
+          long lastElt = outputTicks.get(0);
+          long elt;
+          while (i < outputTicks.size() && descending) {
```

Comment:
keeping track of size in a variable instead of calling size() is probably faster too.
checking from the end and checking if index is > 0 is probably even a tad faster.

Reply:
and not zero is faster, it should be an nz flag in the cpu register :+1:

Reply:
I tried, but cannot go under 7.9s . Obviously, JVM already optimize this.

Reply:
Dorn this Java thing is slow...

Reply:
I tested some variations of this code in a small function, there is no visible improvement. jvm, compiler and branch prediction do a good job.